libpod/oci_conmon_linux.go

```diff
@@ -783,13 +783,31 @@ func (r *ConmonOCIRuntime) ExecContainer(c *Container, sessionID string, options
 	return pid, attachChan, err
 }
 
+// isRetryable returns whether the error was caused by a blocked syscall or the
+// specified operation on a non blocking file descriptor wasn't ready for completion.
+func isRetryable(err error) bool {
+	if errno, isErrno := errors.Cause(err).(syscall.Errno); isErrno {
+		return errno == syscall.EINTR || errno == syscall.EAGAIN
+	}
+	return false
+}
+
 // ExecAttachResize resizes the TTY of the given exec session.
 func (r *ConmonOCIRuntime) ExecAttachResize(ctr *Container, sessionID string, newSize remotecommand.TerminalSize) error {
 	// TODO: probably want a dedicated function to get ctl file path?
 	controlPath := filepath.Join(ctr.execBundlePath(sessionID), "ctl")
-	controlFile, err := os.OpenFile(controlPath, unix.O_WRONLY, 0)
-	if err != nil {
-		return errors.Wrapf(err, "could not open ctl file for terminal resize for container %s exec session %s", ctr.ID(), sessionID)
+	var controlFile *os.File
+	var err error
+
+	for {
+		controlFile, err = os.OpenFile(controlPath, unix.O_WRONLY|unix.O_NONBLOCK, 0)
+		if err == nil {
+			break
+		}
+		if !isRetryable(err) {
+			return errors.Wrapf(err, "could not open ctl file for terminal resize for container %s exec session %s", ctr.ID(), sessionID)
+		}
+		time.Sleep(time.Second / 10)
```

Comment:
Can we do a logrus.info/warning or something here to let the user know we're sleeping?  10 seconds is a long time to wait at a blinking cursor, never mind 60.

Reply:
It is the same logic as before.  It is just to avoid a syscall to block and it happens in a goroutine.

I don't think we need to debug it